econml/cate_estimator.py

```diff
@@ -456,6 +460,93 @@ def const_marginal_effect_inference(self, X=None):
+        ----------
+        X: (m, d_x) matrix
+            Features for each sample. Should be in the same shape of fitted X in final stage.
+        feature_names: optional None or list of strings of length X.shape[1] (Default=None)
+            The names of input features.
+        treatment_names: optional None or list (Default=None)
+            The name of treatment. In discrete treatment scenario, the name should not include control name.
+        output_names:  optional None or list (Default=None)
+            The name of the outcome.
+
+        Returns
+        -------
+        shap_outs: nested dictionary of Explanation object
+            A nested dictionary by using each Y and each T as key and the shap_values explanation object as value.
+
+
+        """
+        d_t = self._d_t[0] if self._d_t else 1
+        d_y = self._d_y[0] if self._d_y else 1
+        if treatment_names is None:
+            treatment_names = [f"T{i}" for i in range(d_t)]
+        if output_names is None:
+            output_names = [f"Y{i}" for i in range(d_y)]
+
+        return _shap_explain_cme(self.const_marginal_effect, X, d_y, d_t, feature_names, treatment_names, output_names)
+
+    def _shap_values(self, models, X, *, feature_names=None, treatment_names=None, output_names=None):
+        """ Shap value for the final stage models.
+        This is the private method mainly for internal use.
+
+        Parameters
+        ----------
+        models: a single estimator or a list of estimators with one estimator per treatment
+            models for the model's final stage model.
+        X: (m, d_x) matrix
+            Features for each sample. Should be in the same shape of fitted X in final stage.
+        feature_names: optional None or list of strings of length X.shape[1] (Default=None)
+            The names of input features.
+        treatment_names: optional None or list (Default=None)
+            The name of treatment. In discrete treatment scenario, the name should not include control name.
+        output_names:  optional None or list (Default=None)
+            The name of the outcome.
+
+        Returns
+        -------
+        shap_outs: nested dictionary of Explanation object
+            A nested dictionary by using each Y and each T as key and the shap_values explanation object as value.
+        """
+
+        d_t = self._d_t[0] if self._d_t else 1
+        d_y = self._d_y[0] if self._d_y else 1
+        if treatment_names is None:
+            treatment_names = [f"T{i}" for i in range(d_t)]
+        if output_names is None:
+            output_names = [f"Y{i}" for i in range(d_y)]
+
+        if not isinstance(models, list):
+            models = [models]
+        assert len(models) == d_t, "Number of final stage models don't equals to number of treatments!"
+        # define masker by using entire dataset, otherwise Explainer will only sample 100 obs by default.
+        background = shap.maskers.Independent(X, max_samples=X.shape[0])
```

Comment:
Why not use the default max_samples as in shap? Wouldn't this make it much slower at no big quality improvement? Why does shap use max_samples=100. 

Also maybe this could be an input to the shap_values function then if it's something that the user should be able to control.

Reply:
I don't think this will slower the computation much, since it's used to compute `base_values` (mean of prediction), the reason I change it since I also want to compare the `base_value` with our mean of `const_marginal_effect` in test to make sure we extract each Y and T correctly